File: 351004/Student/Lab6/publisher/src/main/java/com/example/task310/repository/WriterRepository.java
package com.example.task310.repository;

import com.example.task310.entity.Writer;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface WriterRepository extends JpaRepository<Writer, Long> {
    Optional<Writer> findByLogin(String login);
}

File: 351003/Soika/entities-app/publisher/src/main/kotlin/com/example/entitiesapp/repository/WriterRepository.kt
package com.example.entitiesapp.repository

import com.example.entitiesapp.model.Writer
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface WriterRepository : JpaRepository<Writer, Long> {
    fun findByLogin(login: String): Writer?
}